src/sql/platform/connection/common/connectionConfig.ts

```diff
@@ -214,6 +215,25 @@ export class ConnectionConfig {
 			return ConnectionProfile.createFromStoredProfile(p, this._capabilitiesService);
 		});
 
+		const sortBy = this.configurationService.getValue<string>(CONNECTION_SORT_CONFIG_KEY);
+		let sortFunc: (a: ConnectionProfile, b: ConnectionProfile) => number;
+
+		if (sortBy === 'by title alphabetically') {
+			sortFunc = ((a, b) => {
+				if (a.title < b.title) {
+					return -1;
+				} else if (a.title > b.title) {
+					return 1;
+				} else {
+					return 0;
+				}
+			});
+		}
+
+		if (sortFunc) {
```

Comment:
Could just call this directly in the if check if the sortBy setting is by title

Reply:
I put it this way mainly for potential new sort options other than by title so we can add checks for those in this way. But if `by title` is the only sort option in foreseeable future then sure we can call it directly.

Reply:
Yeah that's fair - it's not a big deal so if you want to leave it like this that's fine. I generally prefer going for the simplest/easiest to read option unless we have actual plans to add others (which I don't believe we have) but go with whatever makes the most sense to you.

Reply:
I'll leave it as it is to allow for other potential options.